### PR TITLE
Bp 0241 edit remove esper queries

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
Extended the EventTypeEndpoint to also offer routes to delete EventTypes.
Also added a new endpoint (EventQueryEndpoint) which is capable of updating existing event queries.

### CODE REVIEW

File: src/main/java/de/hpi/bpt/argos/api/eventQueries/EventQueryEndpointImpl.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * This is the implementation.
+ */
+public class EventQueryEndpointImpl extends RestEndpointImpl implements EventQueryEndpoint {
+
+	@Override
+	public void setup(ResponseFactory responseFactory, PersistenceEntityManager entityManager, Service sparkService) {
+		super.setup(responseFactory, entityManager, sparkService);
+		sparkService.post(EventQueryEndpoint.getUpdateEventQueryBaseUri(), this::updateEventQuery);
+	}
+
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/api/eventQueries/EventQueryEndpointImpl.java:25]: Please inherit the docu

File: src/main/java/de/hpi/bpt/argos/api/eventQueries/EventQueryEndpointImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import spark.Request;
+import spark.Response;
+import spark.Service;
+
+/**
+ * {@inheritDoc}
+ * This is the implementation.
+ */
+public class EventQueryEndpointImpl extends RestEndpointImpl implements EventQueryEndpoint {
+
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/api/eventQueries/EventQueryEndpointImpl.java:16]: Please inherit the docu

File: src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	public NullRestRequest() {
+		response = "null request";
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setMethod(String method) {
+		// empty, since it has no effect
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java:26]: Use ; to indicate a null operation

File: src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void setProperty(String key, String value) {
+		// empty, since it has no effect
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java:34]: Use ; to indicate a null operation

File: src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java [hunk 5]
```diff
@@ -381,6 +382,35 @@ public Event getEvent(long eventId) {
 	 * {@inheritDoc}
 	 */
 	@Override
+	public List<EventQuery> getEventQueries() {
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/persistence/database/DatabaseConnectionImpl.java]: if you don't need this, why don't you remove it?!

File: src/main/java/de/hpi/bpt/argos/persistence/database/PersistenceEntityManagerImpl.java [hunk 6]
```diff
@@ -180,6 +180,14 @@ public Event getEvent(long eventId) {
 	 * {@inheritDoc}
 	 */
 	@Override
+	public List<EventQuery> getEventQueries() {
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/persistence/database/PersistenceEntityManagerImpl.java]: remove pls

File: src/main/java/de/hpi/bpt/argos/persistence/database/PersistenceEntityRetriever.java [hunk 7]
```diff
@@ -91,4 +92,10 @@
 	 * @return - the requested event
 	 */
 	Event getEvent(long eventId);
+
+	/**
+	 * This method makes the database call to retrieve all event queries.
+	 * @return - a list of all event queries
+	 */
+	List<EventQuery> getEventQueries();
```
Comment (tommartensen) [inline src/main/java/de/hpi/bpt/argos/persistence/database/PersistenceEntityRetriever.java]: remove pls

**Review by tommartensen — CHANGES_REQUESTED** [review]

File: src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java (hunk 3 above)
Comment (Johannes-Schneider) [reply src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java:26]: This is not compatible with our style guides.

File: src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java (hunk 4 above)
Comment (Johannes-Schneider) [reply src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java:34]: This is not compatible with our style guides.

Comment (tommartensen) [reply src/main/java/de/hpi/bpt/argos/common/NullRestRequest.java:34]: I found it on this great page: http://userpage.fu-berlin.de/~ram/pub/pub_jf47ht81Ht/java_leere_anweisung_de !

**Pull request merged**